src/main/java/de/teamlapen/vampirism/client/gui/VampirismHUDOverlay.java

```diff
@@ -44,303 +46,332 @@
-                GlStateManager.disableBlend();
-                GlStateManager.enableAlpha();
-                GlStateManager.enableTexture2D();
+	private final Minecraft mc;
+	private final ResourceLocation icons = new ResourceLocation(REFERENCE.MODID + ":textures/gui/icons.png");
+
+	private int screenColor = 0;
+	private int screenPercentage = 0;
+	private boolean fullScreen = false;
+	private int renderFullTick = 0;
+	private int rederFullOn, renderFullOff, renderFullColor;
+	private int screenBottomColor = 0;
+	private int screenBottomPercentage = 0;
+
+	public VampirismHUDOverlay(Minecraft mc) {
+		this.mc = mc;
+	}
+
+	public void makeRenderFullColor(int on, int off, int color) {
+		this.rederFullOn = on;
+		this.renderFullOff = off;
+		this.renderFullTick = on + off;
+		if ((color >> 24 & 255) == 0) {
+			color |= 0xFF000000;
+		}
+		this.renderFullColor = color;
+	}
+
+	@SubscribeEvent
+	public void onClientTick(TickEvent.ClientTickEvent event) {
+		if (mc.player == null)
+			return;
+		if (event.phase == TickEvent.Phase.END)
+			return;
+		IFactionPlayer player = FactionPlayerHandler.get(mc.player).getCurrentFactionPlayer();
+		if (player != null && player instanceof IVampirePlayer) {
+			if (((IVampirePlayer) player).getActionHandler().isActionActive(VampireActions.vampire_rage)) {
+				screenPercentage = 100;
+				screenColor = 0xfff00000;
+				fullScreen = false;
+			} else if ((screenPercentage = ((IVampirePlayer) player).getTicksInSun() / 2) > 0) {
+				PotionEffect effect = mc.player.getActivePotionEffect(ModPotions.sunscreen);
+				if (effect == null || effect.getAmplifier() < 5) {
+					screenColor = 0xfffff755;
+					fullScreen = false;
+					if (player.getRepresentingPlayer().capabilities.isCreativeMode) {
+						screenPercentage = Math.min(20, screenPercentage);
+					}
+				} else {
+					screenPercentage = 0;
+				}
+			}
+			setColorForBatDuration(player);
+
+		} else if (player != null && player instanceof HunterPlayer && ((HunterPlayer) player).getSpecialAttributes().isDisguised()) {
+			screenPercentage = (int) (100 * ((HunterPlayer) player).getSpecialAttributes().getDisguiseProgress());
+			screenColor = 0xff111111;
+			fullScreen = false;
+		} else {
+			screenPercentage = 0;
+		}
+		if (renderFullTick > 0) {
+			screenColor = renderFullColor;
+			fullScreen = true;
+			if (renderFullTick > renderFullOff) {
+				screenPercentage = (int) (100 * (1 - (renderFullTick - renderFullOff) / (float) rederFullOn));
+			} else {
+				screenPercentage = (int) (100 * renderFullTick / (float) renderFullOff);
+			}
+			renderFullTick--;
+		}
+
+	}
+
+	private void setColorForBatDuration(IFactionPlayer player) {
+		float batPercentage = ((IVampirePlayer) player).getActionHandler().getPercentageForAction(VampireActions.bat);
+		if (batPercentage < 0.1F && batPercentage > 0.0F) {
+			screenBottomColor = 0xcc7067f9; // change color
+			screenBottomPercentage = (int) (0.1F - batPercentage * 10);
```

Comment:
I assume screenBottomPercentage should go from 0 to 100. Since batPercentage can have a range between 0.1 and 0). So I would propose changing this to `(int) (100.0F*10*(0.1F - batPercentage))`